### PR TITLE
Fix `host info` for host with multiple IPs on same MAC

### LINKED ISSUE #337: Looking up a host by MAC address should check if multiple MACs resolve to the same host.
If a host has multiple IPs and they are associated to the same MAC we can get:
```
foo@bar> host info 50:9a:4c:46:1e:6b
Expected exactly one result, got 2.
```

### PR DESCRIPTION
**Pull request by pederhan**
Fixes #337 

We should probably have a test in the test suite for this.

### CODE REVIEW

File: mreg_cli/utilities/api.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def get_list_generic(
     if expect_one_result:
         if len(ret) == 0:
             return {}
-        if len(ret) != 1:
+        if len(ret) > 1 and not all(ret[0] == x for x in ret):
             raise MultipleEntititesFound(f"Expected exactly one result, got {len(ret)}.")
```
[the remark this answers is not in the document]
Comment (pederhan) [reply mreg_cli/utilities/api.py:544]: Yeah, can do.

[the remark this answers is not in the document]
Comment (pederhan) [reply mreg_cli/utilities/api.py:544]: Well, we don't know that they are _all_ distinct from each other, but I can't think of a good message that conveys the ambiguity involved in interpreting the results.

File: mreg_cli/utilities/api.py [hunk 2]
```diff
@@ -540,7 +540,7 @@ def get_list_generic(
     if expect_one_result:
         if len(ret) == 0:
             return {}
-        if len(ret) != 1:
+        if len(ret) > 1 and not all(ret[0] == x for x in ret):
```
Comment (oyvindkolbu) [inline mreg_cli/utilities/api.py:543]: Suggest changing `not all` to `any` and thus the `==` to `!=`.

File: mreg_cli/utilities/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (terjekv) [reply mreg_cli/utilities/api.py]: We could theoretically check for the number of truly distinct matches, but I'm not sure that'd make things any clearer. :)

File: mreg_cli/utilities/api.py (hunk 2 above)
Comment (pederhan) [reply mreg_cli/utilities/api.py:543]: Yes, that short circuits faster, you are right.

File: ci/testsuite-result.json [hunk 3]
```diff
@@ -20022,6 +20022,127 @@
...
+    "ok": [],
+    "warning": [],
+    "error": [],
+    "output": [
+      "Name:         foo.example.org",
+      "Contact:      ",
+      "A_Records     IP            MAC",
+      "              10.0.0.5      aa:bb:cc:dd:ee:ff",
+      "AAAA_Records  IP            MAC",
+      "              2001:db8::5   <not set>",
```
Comment (pederhan) [inline ci/testsuite-result.json:20039]: Need to fix MAC association here

Comment (terjekv) [reply ci/testsuite-result.json:20039]: Do we? I mean, it's implied due to the way things work at UiO, but the association is not explicitly set and I don't think that was the intent.

Comment (pederhan) [reply ci/testsuite-result.json:20039]: Ok, let's leave it as it is.

File: mreg_cli/api/models.py [hunk 4]
```diff
@@ -2701,7 +2701,7 @@ def get_by_any_means(
                     return hosts[0]
 
                 if len(hosts) > 1:
-                    raise MultipleEntititesFound(
+                    raise MultipleEntitiesFound(
```
Comment (pederhan) [inline mreg_cli/api/models.py:2704]: Could we hit the same error here?

File: mreg_cli/api/models.py [hunk 5]
```diff
@@ -2701,7 +2701,7 @@ def get_by_any_means(
...
                         f"Multiple hosts found with IP address or PTR {identifier}."
```
Comment (pederhan) [inline mreg_cli/api/models.py:2705]: Could we hit the same error here?

Comment (terjekv) [reply mreg_cli/api/models.py:2705]: Yes, indeed.

File: mreg_cli/api/models.py [hunk 6]
```diff
@@ -2701,7 +2701,7 @@ def get_by_any_means(
                     return hosts[0]
 
                 if len(hosts) > 1:
```
Comment (pederhan) [inline mreg_cli/api/models.py:2703]: We could do this:

```suggestion
                if len(hosts) > 1 and any(hosts[0].id != host.id for host in hosts):
```

But should we consider refactoring the method used to fetch hosts to `Host.get_list_by_query_unique`, and then instead catch `MultipleEntitiesFound` and raise the more granular `MultipleEntitiesFound` we already have? Something like:


```py
            try:
                ptr = False
                ipaddress.ip_address(identifier)

                host = Host.get_by_query_unique({"ipaddresses__ipaddress": identifier})
                if not host:
                    host = Host.get_by_query_unique({"ptr_overrides__ipaddress": identifier})
                    ptr = True

                if host:
                    if ptr and inform_as_ptr:
                        OutputManager().add_line(f"{identifier} is a PTR override for {host.name}")
                    return host
            except MultipleEntitiesFound as e:
                raise MultipleEntitiesFound(
                    f"Multiple hosts found with IP address or PTR {identifier}."
                ) from e
            except ValueError: # invalid IP (should use NetworkOrIP.parse...)
                pass
```

Comment (pederhan) [reply mreg_cli/api/models.py]: I have pushed a commit which implements this. I do think we need more comprehensive tests for a broader range of host/ip/mac/network permutations though. I'll look into it once we have implemented #323.

**Review by terjekv — APPROVED** [review]
LGTM

**Pull request merged**